server/src/main/java/org/candlepin/auth/permissions/JobStatusPermission.java

```diff
@@ -99,11 +97,21 @@ public boolean canAccessTarget(JobStatus target, SubResource subResource, Access
         }
 
         if ("user".equalsIgnoreCase(principalType)) {
-            // Org was OK, but only allow a user to view consumer's status.
-            return true;
+            // A user can only cancel the job if it started it
+            if (principalNameMatch) {
+                return Access.ALL.provides(required);
+            }
+            else {
+                return Access.READ_ONLY.provides(required);
+            }
+        }
+        else if ("consumer".equalsIgnoreCase(principalType)) {
+            // if this is a consumer it must match the principle name access
+            return Access.ALL.provides(required) && principalNameMatch;
+        }
+        else {
```

Comment:
Being a little picky, but you can save a couple of lines since the last else is not required -- you can just return false.